api.planx.uk/send/bops.js

```diff
@@ -1,27 +1,37 @@
-import {
-  responseInterceptor,
-  fixRequestBody,
-} from "http-proxy-middleware";
-import { adminGraphQLClient } from "../hasura";
+import { responseInterceptor } from "http-proxy-middleware";
+import { adminGraphQLClient as client } from "../hasura";
 import { useProxy } from "../server";
 import { markSessionAsSubmitted } from "../saveAndReturn/utils";
 
-const client = adminGraphQLClient;
+const sendToBOPS = async (req, res, next) => {
+  // `/bops/:localAuthority` is only called via Hasura's scheduled event webhook now, so body is wrapped in a "payload" key
+  const { payload } = req.body;
+  if (!payload) {
+    next({
+      status: 400,
+      message: `Missing application payload data to send to BOPS`,
+    });
+  }
+
+  // confirm that this session has not already been successfully submitted before proceeding
+  const submittedApp = await checkBOPSAuditTable(payload?.planx_debug_data?.session_id);
```

Comment:
nit: For a future PR! I'm also wrote a function which relied on `payload?.planx_debug_data?.session_id` for BOPS - I think these two functions basically confirm that it's a core part of the process and shouldn't be buried in the "temporary" (?) debug data.